UnityProject/Assets/Scripts/Core/Lighting/LightsHolder.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Linq;
+using Light2D;
+using Mirror;
+using UnityEngine;
+
+namespace Core.Lighting
+{
+	public class LightsHolder : NetworkBehaviour
+	{
+		public SyncList<LightData> Lights { get; private set; } = new SyncList<LightData>();
+
+		[SerializeField] private Transform lightsParent;
+		[SerializeField] private GameObject lightSpriteObject;
+		[SerializeField] private Rotatable rotatable;
+
+		private OrientationEnum currentOrientation = OrientationEnum.Default;
+
+		private void Start()
+		{
+			rotatable ??= GetComponent<Rotatable>();
+			UpdateLights();
+			rotatable.OnRotationChange.AddListener(OnRotate);
+		}
+
+		private void OnDestroy()
+		{
+			rotatable.OnRotationChange.RemoveListener(OnRotate);
+		}
+
+		private void OnRotate(OrientationEnum orientation)
+		{
+			currentOrientation = orientation;
+			//We add 90 to deal with the texture's own origin/pivot.
+			var rotation = Quaternion.Euler(0f, 0f, (float)orientation * 90 + 90);
+			foreach (Transform child in lightsParent)
+			{
+				child.rotation = rotation;
+			}
+		}
+
+		public void AddLight(LightData data)
+		{
+			if (Lights.Any(lightSprite => lightSprite.Id == data.Id))
+			{
+				return;
+			}
+			Lights.Add(data);
+			UpdateLights();
+		}
+
+		public void RemoveLight(LightData data)
+		{
+			if (Lights.Contains(data) == false)
+			{
+				Logger.LogError("Could not find correct light source to remove.");
+				ClearHeldLights();
+				UpdateLights();
+				return;
+			}
+
+			foreach (var sprite in lightsParent.GetComponentsInChildren<LightSprite>())
+			{
+				if (sprite.GivenID == data.Id) Despawn.ClientSingle(sprite.gameObject);
+			}
+
+			Lights.Remove(data);
+			UpdateLights();
+		}
+
+		[ClientRpc]
```

Comment:
there's a hook for  SyncList so you don't need to manually call, and also re-logging works with hook